### PR TITLE
PMM-9197 migrate to hcl packer format

### PR DESCRIPTION
**Pull request by nikita-b**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,31 +43,22 @@ pmm-az:
 	packer build -only azure-arm packer/pmm.json
 
 pmm2-ovf: fetch
-	packer build -only virtualbox-ovf packer/pmm2.json
+	packer build -only virtualbox-ovf.ovf_distribuion packer/pmm2.pkr.hcl
```
Comment (atymchuk) [inline Makefile:46]:
```suggestion
	packer build -only virtualbox-ovf.ovf_distribution packer/pmm2.pkr.hcl
```

File: Makefile [hunk 2]
```diff
@@ -43,31 +43,22 @@ pmm-az:
...
 
 pmm2-digitalocean:
-	packer build -only digitalocean -var 'single_disk=true' packer/pmm2.json
+	packer build -only digitalocean.digitalocean_distribuion -var 'single_disk=true' packer/pmm2.pkr.hcl
```
Comment (atymchuk) [inline Makefile:49]:
```suggestion
	packer build -only digitalocean.digitalocean_distribution -var 'single_disk=true' packer/pmm2.pkr.hcl
```

File: Makefile [hunk 3]
```diff
@@ -43,31 +43,22 @@ pmm-az:
...
+	packer build -var 'pmm_version=2.25.0' -only azure-arm.azure_distribuion packer/pmm2.pkr.hcl
 
 pmm2-ami:
 	docker run --rm -v ${HOME}/.aws:/root/.aws -v `pwd`:/build -w /build hashicorp/packer:${PACKER_VERSION} \
 			build -var 'pmm_client_repos=original experimental' \
                   -var 'pmm_client_repo_name=percona-experimental-x86_64' \
                   -var 'pmm2_server_repo=experimental' \
-                  -only amazon-ebs -color=false \
-				  packer/pmm2.json
+                  -only amazon-ebs.aws_distribuion -color=false \
```
Comment (atymchuk) [inline Makefile:59]:
```suggestion
                  -only amazon-ebs.aws_distribution -color=false \
```

File: Makefile [hunk 4]
```diff
@@ -43,31 +43,22 @@ pmm-az:
...
 
 pmm2-azure:
-	packer build -only azure-arm packer/pmm2.json
-
-docker-ovf: fetch
-	packer build -only virtualbox-ovf packer/docker.json
-
-centos-ami:
-	packer build packer/centos-ami.json
+	packer build -var 'pmm_version=2.25.0' -only azure-arm.azure_distribuion packer/pmm2.pkr.hcl
```
Comment (atymchuk) [inline Makefile:52]:
```suggestion
	packer build -var 'pmm_version=2.25.0' -only azure-arm.azure_distribution packer/pmm2.pkr.hcl
```

File: Makefile [hunk 5]
```diff
@@ -43,31 +43,22 @@ pmm-az:
...
+                  -only amazon-ebs.aws_distribuion -color=false \
+				  packer/pmm2.pkr.hcl
 pmm2-ami-rc:
 	docker run --rm -v ${HOME}/.aws:/root/.aws -v `pwd`:/build -w /build hashicorp/packer:${PACKER_VERSION} \
-			build -var 'pmm_client_repos=original testing' \
-				  -var 'pmm_client_repo_name=percona-testing-x86_64' \
-				  -var 'pmm2_server_repo=testing' \
-				  -only amazon-ebs '-color=false' \
-				  packer/pmm2.json
+			build -only amazon-ebs.aws_distribuion '-color=false' \
```
Comment (atymchuk) [inline Makefile:63]:
```suggestion
			build -only amazon-ebs.aws_distribution '-color=false' \
```

File: packer/pmm2.pkr.hcl [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  type    = string
+  default = "original testing"
+}
+
+variable "single_disk" {
+  type    = string
+  default = "false"
+}
+
+source "amazon-ebs" "aws_distribuion" {
```
Comment (atymchuk) [inline packer/pmm2.pkr.hcl:22]:
```suggestion
source "amazon-ebs" "aws_distribution" {
```

File: packer/pmm2.pkr.hcl [hunk 7]
```diff
@@ -0,0 +1,192 @@
+
+variable "pmm2_server_repo" {
+  type    = string
+  default = "testing"
+}
+
+variable "pmm_client_repo_name" {
+  type    = string
+  default = "percona-testing-x86_64"
```
Comment (atymchuk) [inline packer/pmm2.pkr.hcl:9]: Is it a real repository name? I couldn't find it )

File: packer/pmm2.pkr.hcl [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+    iit-billing-tag = "pmm-ami"
+  }
+  security_group_id = "sg-688c2b1c"
+  source_ami        = "ami-00e87074e52e6c9f9"
+  ssh_pty           = "true"
+  ssh_username      = "centos"
+  subnet_id         = "subnet-ee06e8e1"
+}
+
+source "azure-arm" "azure_distribuion" {
```
Comment (atymchuk) [inline packer/pmm2.pkr.hcl:52]:
```suggestion
source "azure-arm" "azure_distribution" {
```

File: packer/pmm2.pkr.hcl [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+  ssh_username         = "vagrant"
+  vboxmanage           = [["modifyvm", "{{ .Name }}", "--memory", "4096"],
+                          ["modifyvm", "{{ .Name }}", "--audio", "none"],
+                          ["createhd", "--format", "VMDK", "--filename", "/tmp/{{ .Name }}-disk2.vmdk", "--variant", "STREAM", "--size", "409600"], ["storagectl", "{{ .Name }}", "--name", "SCSI Controller", "--add", "scsi", "--controller", "LSILogic"], ["storageattach", "{{ .Name }}", "--storagectl", "SCSI Controller", "--port", "1", "--type", "hdd", "--medium", "/tmp/{{ .Name }}-disk2.vmdk"]]
+  vm_name              = "PMM2-Server-${formatdate("2006-01-02 1504", timestamp())}"
+}
+
+build {
+  sources = [
+    "source.amazon-ebs.aws_distribuion",
```
Comment (atymchuk) [inline packer/pmm2.pkr.hcl:109]: Need to change the spelling here as well

**Review by atymchuk — CHANGES_REQUESTED** [review]

File: packer/pmm2.pkr.hcl (hunk 7 above)
Comment (nikita-b) [reply packer/pmm2.pkr.hcl:9]: Yep!
It's testing repo:
```
[root@459f63962611 opt]# grep -r percona-testing-x86_64 /etc/yum.repos.d/
/etc/yum.repos.d/percona-original-testing.repo.bak:[percona-testing-x86_64]
```

**Review by atymchuk — APPROVED** [review]

**Pull request closed without merging**